src/library_manager/lib_manager.c

```diff
@@ -54,7 +54,7 @@ static int lib_manager_load_data_from_storage(void __sparse_cache *vma, void *s_
 					      uint32_t size, uint32_t flags)
 {
 	int ret = sys_mm_drv_map_region((__sparse_force void *)vma, POINTER_TO_UINT(NULL),
-					size, flags);
+					size, SYS_MM_MEM_PERM_RW);
```

Comment:
looks wrong. You make an effort calculating flags below and here you overwrite them

Reply:
A region must be first mapped as writable in order to initialize its contents. Next, the expected access flags should be set (todo on line 67), but currently the tlb driver does not support this. In fact, the tlb driver currently completely ignores access flags and maps all regions as rwx.

Reply:
ok, this is probably worth commenting in the code to avoid misunderstandings.